test/extensions-tests.js

```diff
@@ -246,8 +246,22 @@ describe('extensions', function () {
     it.skip('behaves like then except for not returning anything', function () {
       //todo
     })
-    it.skip('rethrows unhandled rejections', function () {
-      //todo
+    it ('rethrows unhandled rejections', function (done) {
+      var originalTimeout = global.setTimeout
+
+      global.setTimeout = function(callback) {
+        try {
+          callback()
+        } catch (x) {
+          assert(x.message === 'It worked')
+        }
+        global.setTimeout = originalTimeout
+        done()
```

Comment:
This looks good, except that you aren't actually checking that `callback` throws an exception.  You should move the cleanup code into the `catch` handler, then add a `return` statement inside the catch handler and a `throw` statement after the catch handler that's something like `throw new Error('Callback should have thrown an exception');`